docs: Update drift docs to remove drift concepts (#4436)
aws/karpenter: website/content/en/preview/concepts/deprovisioning.md

Reviewer: Can you add to docs folder too?

## Drift

> Drift triggers Karpenter to roll your nodes when the spec of the owning Provisioner changes. Some special cases will be reconciled two-ways, where resolved values are dynamically discovered and changes to the owning Provisioner or NodeTemplate may or may not actually trigger drift on the nodes based on whether the resolved value on the Machine fits within the owning object's dynamically discovered constraints.

Reviewer 2: ```suggestion
Drift triggers Karpenter to roll your nodes when the spec of the owning Provisioner changes. Some special cases will be reconciled two-ways, where resolved values are dynamically discovered. This means that changes to the owning Provisioner or NodeTemplate may or may not actually trigger drift based on whether the resolved value on the Machine fits within the owning object's dynamically discovered constraints.
```

Realized after reading that that sentence is quite lengthy

> Most fields are only triggered by CRD changes. Some special cases will be reconciled two-ways, triggered by machine/node/instance changes and Provisioner or AWSNodeTemplate changes. For one-way reconciliation, values in the CRDs are reflected in the machine in the same way that they’re set. A machine will be detected as drifted if the values in the CRDs do not match the values in the machine. By default, fields are drifted using one-way reconciliation. 

Reviewer 2: ```suggestion
Drift triggers Karpenter to roll your nodes when the spec of the owning Provisioner changes. Some special cases will be reconciled two-ways, where resolved values are dynamically discovered and changes to the owning Provisioner or NodeTemplate may or may not actually trigger drift on the nodes based on whether the resolved value on the Machine fits within the owning object's dynamically discovered constraints.
```

Some of this feels lengthy. I wonder if we can collapse some of this wording down.

> Most fields are only triggered by CRD changes. Some special cases will be reconciled two-ways, triggered by machine/node/instance changes and Provisioner or AWSNodeTemplate changes. For one-way reconciliation, values in the CRDs are reflected in the machine in the same way that they’re set. A machine will be detected as drifted if the values in the CRDs do not match the values in the machine. By default, fields are drifted using one-way reconciliation. 

Reviewer 2: ```suggestion
Drift on most fields are only triggered by changes to the owning CustomResource. Some special cases will be reconciled two-ways, triggered by Machine/Node/Instance changes or Provisioner/AWSNodeTemplate changes. For one-way reconciliation, values in the CustomResource are reflected in the Machine in the same way that they’re set. A machine will be detected as drifted if the values in the CRDs do not match the values in the Machine. By default, fields are drifted using one-way reconciliation. 
```

The start of this section doesn't flow well to me IMO.

> Most felids are implemented one-way reconciliation, triggered only by CRD changes. Some special cases will be a two-way reconciled, triggered by machine/node/instance changes and Provisioner or AWSNodetemplate changes. For One-way Reconciliation, values in the CRDs are reflected in the machine in the same way that they’re set. A machine will be detected as drifted if the values in the CRDs do not match the values in the machine. By default, fields are drifted using One-way Reconciliation. 

Reviewer: some nits on grammar, capitalization. Happy to capitalize one-way or two-way or reconciliation, but should be consistent throughout.
```suggestion
Most fields are only triggered by CRD changes. Some special cases will be reconciled two-ways, triggered by machine/node/instance changes and Provisioner or AWSNodeTemplate changes. For one-way reconciliation, values in the CRDs are reflected in the machine in the same way that they’re set. A machine will be detected as drifted if the values in the CRDs do not match the values in the machine. By default, fields are drifted using one-way reconciliation. 
```

### Two-way Reconciliation

> Two-way reconciliation can correspond to multiple values and must be handled differently. Two-way reconciliation can create cases where drift occurs without changes to CRDs, or where CRD changes do not result in drift. For example, if a machine has `node.kubernetes.io/instance-type: m5.large`, and requirements change from `node.kubernetes.io/instance-type In [m5.large]` to `node.kubernetes.io/instance-type In [m5.large, m5.2xlarge]`, the machine will not be drifted because its value is still compatible with the new requirements. Conversely, for an AWS Installation, if a machine is using a machine image `ami: ami-abc`, but a new image is published, Karpenter's `AWSNodeTemplate.amiSelector` will discover that the new correct value is `ami: ami-xyz`, and detect the machine as drifted.

Reviewer 2: ```suggestion
Two-way reconciliation can create cases where drift occurs without changes to a CRD, or where CRD changes do not result in drift. For example, if a Machine has `node.kubernetes.io/instance-type: m5.large`, and requirements change from `node.kubernetes.io/instance-type In [m5.large]` to `node.kubernetes.io/instance-type In [m5.large, m5.2xlarge]`, the Machine will not be drifted because its value is still compatible with the new requirements. Conversely, if a Machine is using an image `ami: ami-abc`, but a new image is published, Karpenter's `AWSNodeTemplate.amiSelector` will discover that the new correct value is `ami: ami-xyz`, and detect the Machine as drifted.
```

nit: Reduced the opening line to make it more concise

> Two-way reconciliation can correspond to multiple values and must be handled differently. Two-way reconciliation can create cases where drift occurs without changes to CRDs, or where CRD changes do not result in drift. For example, if a machine has `node.kubernetes.io/instance-type: m5.large`, and requirements change from `node.kubernetes.io/instance-type In [m5.large]` to `node.kubernetes.io/instance-type In [m5.large, m5.2xlarge]`, the machine will not be drifted because it's value is still compatible with the new requirements. Conversely, for an AWS Installation, if a machine is using a machine image `ami: ami-abc`, but a new image is published, Karpenter's `AWSNodeTemplate.amiSelector` will discover that the new correct value is `ami: ami-xyz`, and detect the machine as drifted.

Reviewer: sorry this is my mistake!
```suggestion
Two-way reconciliation can correspond to multiple values and must be handled differently. Two-way reconciliation can create cases where drift occurs without changes to CRDs, or where CRD changes do not result in drift. For example, if a machine has `node.kubernetes.io/instance-type: m5.large`, and requirements change from `node.kubernetes.io/instance-type In [m5.large]` to `node.kubernetes.io/instance-type In [m5.large, m5.2xlarge]`, the machine will not be drifted because its value is still compatible with the new requirements. Conversely, for an AWS Installation, if a machine is using a machine image `ami: ami-abc`, but a new image is published, Karpenter's `AWSNodeTemplate.amiSelector` will discover that the new correct value is `ami: ami-xyz`, and detect the machine as drifted.
```

### Behavioral Fields
Behavioral Fields are treated as over-arching settings on the Provisioner to dictate how Karpenter behaves. These fields don’t correspond to settings on the machine or instance. They’re set by the user to control Karpenter’s Provisioning and Deprovisioning logic. Since these don’t map to a desired state of machines, __behavioral fields are not considered for Drift__.

Read the [Drift Design](https://github.com/aws/karpenter-core/blob/main/designs/drift.md) for more.

#### Provisioner

> | Fields                     | One-way | Two-way | 

Reviewer 2: ```suggestion
| Fields                     | One-Way | Two-Way | 
```

- TTLSecondsAfterEmpty  
---      
#### AWSNodeTemplate

> | Fields                     | One-way | Two-way  |

Reviewer 2: ```suggestion
| Fields                     | One-Way | Two-Way  |
```
